lib/hanami/model/migrator/adapter.rb

```diff
@@ -110,6 +121,21 @@ def version
 
         private
 
+        # @since x.x.x
+        # @api private
+        def version_to_rollback(table, steps)
+          record = table.order(Sequel.desc(MIGRATIONS_TABLE_VERSION_COLUMN)).all[steps.abs]
+          return 0 unless record
+
+          record.fetch(MIGRATIONS_TABLE_VERSION_COLUMN).scan(/\A[\d]{14}/).first.to_i
```

Comment:
This has implicit knowledge of the structure of the filename. I bet there are more places where this is known. Could we possible centralise that structural knowledge someplace?
I think it would make this code more resilient to bugs if it didn't care about the file structure at all.